File: python/test/secret/Secret_test.py
import unittest

import numpy

from station.DataStation import DataStation
from secret import generateSecret
from util.Util import matrixDiagonalMultiplication


class SecretTest(unittest.TestCase):
    def test_GenerateSecret(self):
        parties = []
        length = 10;

        dummy = numpy.zeros((length, length))
        for i in range(0,10):
            parties.append(DataStation(i, dummy));


        secret = generateSecret(parties);

        sum = 0
        matrices = []

        for part in secret.getParts():
            self.assertNotEquals(part.getR(), 0)
            sum = sum + part.getR();
            matrices.append(part.getMatrix());


        #Due to machine precision this sometimes fails.
        #This is despites Pythons claim it can work with giant numbers.
        #Same code works fine with BigInteger in java
        self.assertEquals(sum, matrixDiagonalMultiplication(matrices));

File: java/src/test/java/secret/SecretTest.java
package secret;

import org.junit.jupiter.api.Test;
import station.DataStation;

import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;
import static util.Util.matrixDiagonalMultiplication;

public class SecretTest {

    @Test
    public void testGenerateSecret() {
        List<DataStation> parties = new ArrayList<>();
        int length = 10;
        for (int i = 0; i < length; i++) {
            parties.add(new DataStation(String.valueOf(i), new BigInteger[length][length]));
        }
        Secret secret = Secret.generateSecret(parties);

        BigInteger sum = BigInteger.ZERO;
        List<BigInteger[][]> matrices = new ArrayList<>();
        for (SecretPart part : secret.getParts()) {
            assertNotEquals(part.getR(), BigInteger.ZERO);
            sum = sum.add(part.getR());
            matrices.add(part.getMatrix());
        }
        assertEquals(sum, matrixDiagonalMultiplication(matrices, length));

    }
}